packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1153,6 +1166,8 @@ export class GarbageCollector implements IGarbageCollector {
             );
         }
         const gcSnapshotTree = snapshot.trees[gcTreeKey];
+        // If GC ran in the container that generated this snapshot, it will have a GC tree.
+        this.wasGCRunInLatestSummary = gcSnapshotTree !== undefined;
```

Comment:
All these changes make me wonder if we should use some sort of state machine to figure out what we need to do. Maybe a state checker of sorts.

Reply:
Good point. But there are only 2 states here - state needs reset or it doesn't. Adding state machine may be a bit much for it.

Reply:
I guess, is there a way of making wasGCRunInLatestSummary more function like looking at the current properties of GC, so we don't have to maintain as much state? The more state we have the harder it is to make sure everything is in order. Unless making it function like is expensive of course.

Reply:
There isn't another state for this. That's the reason I added this as a state to track and gcStateNeedsReset is a getter function which uses this.